TypeSystem/flow-analysis/promotion_via_type_test_A03_t09.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a variable `x` is promotable via type test with type
+/// `T` given variable model `VM` if
+/// - `VM = VariableModel(declared, promoted, tested, assigned, unassigned,
+///     captured)`
+/// - and `captured` is `false`
+/// - and `S` is the current type of `x` in `VM`
+/// - and not `S <: T`
+/// - and `T <: S` or (`S` is `X extends R` and `T <: R`) or (`S` is `X & R` and
+///   `T <: R`)
+///
+/// @description Checks that a variable is promoted if `T` is a subtype of `S`.
+/// Test type `dynamic` as `T`.
+/// @author sgrekhov22@gmail.com
+
+main() {
+  Object? o = 1 > 2 ? null : "a";
+  if (o is dynamic) { // ignore: unnecessary_type_check
+    try {
+      o.whatever;
```

Comment:
I think we can just expect an error here: There are no other tests in this library, so it's not a problem if we change it from being a "compiles and runs" test into an "encounters compile-time errors" test.

(Or would that imply that the test should have a different name because error tests and non-error tests can somehow be recognized by their names? In that case I guess it could just be renamed).

Reply:
Thanks!